lib/merlin.ml

```diff
@@ -247,53 +242,47 @@ module Cmd = struct
     let* query_cmd =
       match query_type with
       | Query_type.Locate ->
-          let* loc = retrieve_loc ~query_type loc in
-          Result.ok
-          @@ Format.asprintf
-               " %a %s -look-for ml -position '%a' -filename %a < %a" basic_cmd
-               merlin
-               (Query_type.to_string query_type)
-               (Location.print_edge Right)
-               loc File.pp file File.pp file
+          let+ loc = retrieve_loc ~query_type loc in
+          Format.asprintf " %a %s -look-for ml -position '%a' -filename %a < %a"
+            basic_cmd merlin
+            (Query_type.to_string query_type)
+            (Location.print_edge Right)
+            loc File.pp file File.pp file
```

Comment:
Lol. Good catch! xD

Reply:
Yes, in many case, I still prefer the combo `let*` + `return` if there is more than one case but here I guess that it is only for result deconstruction so I prefer remove a level of indentation.